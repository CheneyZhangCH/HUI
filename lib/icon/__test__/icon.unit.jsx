import * as renderer from 'react-test-renderer';
import React from 'react';
import { mount } from 'enzyme';
// renderer 测内容  mount 测事件
import Icon from '../icon';

describe('Icon', () => {
  it('should be svg ', () => {
    // const tree = renderer
    //   .create(<Button icon="alipay"><span>Click Me</span></Button>)
    //   .toJSON()
    // expect(tree).toMatchSnapshot()
    const json = renderer.create(<Icon/>).toJSON();
    expect(json).toMatchSnapshot();
  });
  it('fn 被调用了', () => {
    const fn = jest.fn();
    const c = mount(<Icon name='wechat' onClick={fn}/>);
    expect(c.find('use').simulate('click'));
    expect(fn).toBeCalled();
  });
});