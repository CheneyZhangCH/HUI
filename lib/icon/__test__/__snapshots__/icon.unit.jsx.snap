// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should be svg  1`] = `
<svg
  className="hui-icon"
>
  <use
    xlinkHref="#undefined"
  />
</svg>
`;
